### PR TITLE
fix service page when the href is link to split toc

### PR DESCRIPTION
**Pull request by AngryBerryMS**
[AB#344232](https://dev.azure.com/ceapex/Engineering/_workitems/edit/344232)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6862)

### CODE REVIEW

File: src/docfx/build/toc/ServicePageGenerator.cs [hunk 1]
```diff
@@ -111,13 +117,14 @@ public void GenerateServicePageFromTopLevelTOC(TocNode node, List<FilePath> resu
...
                     var childUid = item.Value.Uid.Value;
                     var childHrefType = TocLoader.GetHrefType(childHref);
-                    if (!string.IsNullOrEmpty(childHref) && (childHrefType == TocHrefType.RelativeFolder || childHrefType == TocHrefType.TocFile))
-                    {
-                        childHref = null;
-                    }
 
                     if (item.Value.LandingPageType.Value != null)
                     {
+                        if (!string.IsNullOrEmpty(childHref) && (childHrefType == TocHrefType.RelativeFolder || childHrefType == TocHrefType.TocFile))
```
Comment (live1206) [inline src/docfx/build/toc/ServicePageGenerator.cs:123]:
```suggestion
                        if (!string.IsNullOrEmpty(childHref)) 
                        {
                          if (childHrefType == TocHrefType.RelativeFolder || childHrefType == TocHrefType.TocFile) 
                          {
                            childHref = null;
                          }
                          else 
                          {
                            childHref = GetHrefRelativeToServicePage(childHref, referenceTOCFullPath, servicePagePath);
                          }
                        }
```

File: src/docfx/build/context/Input.cs [hunk 2]
```diff
@@ -78,6 +78,22 @@ public bool Exists(FilePath file)
             return package.Exists(path);
         }
 
+        public bool ExistInGeneratedContents(string pathString, out FilePath? filePath)
```
[the remark this answers is not in the document]
Comment (AngryBerryMS) [reply src/docfx/build/context/Input.cs]: Fixed.

File: src/docfx/build/toc/ServicePageGenerator.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/build/toc/ServicePageGenerator.cs]: Fixed.

**Review by live1206 — APPROVED** [review]

**Pull request merged**